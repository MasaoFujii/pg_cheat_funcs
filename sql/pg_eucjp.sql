CREATE DATABASE regtest_cheat_funcs_eucjp ENCODING 'EUC_JP' TEMPLATE template0;
\c regtest_cheat_funcs_eucjp

SET client_encoding TO 'EUC_JP';

CREATE EXTENSION pg_cheat_funcs;

\pset null '(null)'

SELECT pg_eucjp(65::bit(8));
SELECT pg_eucjp(65::bit(8), 65::bit(8));

SELECT pg_eucjp('xa4', 'xca');
SELECT pg_eucjp('xa4', 'xca', 'xca');
SELECT pg_eucjp('xa0', 'xca');
SELECT pg_eucjp('xa4', 'xff');

SELECT pg_eucjp('x8e', 'xc5');
SELECT pg_eucjp('x8e', 'xc5', 'xc5');
SELECT pg_eucjp('x8e', 'xe0');

SELECT pg_eucjp('x8f', 'xa2', 'xee');
SELECT pg_eucjp('x8f', 'xa0', 'xee');
SELECT pg_eucjp('x8f', 'xa2', 'xff');

CREATE TABLE all_eucjp(code1 text, code2 text, code3 text, eucjp text);
INSERT INTO all_eucjp SELECT * FROM pg_all_eucjp();
SELECT count(*) FROM all_eucjp
    WHERE pg_eucjp(code1::bit(8), code2::bit(8), code3::bit(8)) <> eucjp;
SELECT * FROM all_eucjp ORDER BY eucjp;

DROP EXTENSION pg_cheat_funcs;

\c contrib_regression
DROP DATABASE regtest_cheat_funcs_eucjp;
