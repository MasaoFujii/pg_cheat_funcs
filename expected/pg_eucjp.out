CREATE DATABASE regtest_cheat_funcs_eucjp ENCODING 'EUC_JP' TEMPLATE template0;
ERROR:  database "regtest_cheat_funcs_eucjp" already exists
\c regtest_cheat_funcs_eucjp
SET client_encoding TO 'EUC_JP';
CREATE EXTENSION pg_cheat_funcs;
\pset null '(null)'
SELECT pg_eucjp(65::bit(8));
 pg_eucjp 
----------
 A
(1 row)

SELECT pg_eucjp(65::bit(8), 65::bit(8));
 pg_eucjp 
----------
 (null)
(1 row)

SELECT pg_eucjp('xa4', 'xca');
 pg_eucjp 
----------
 § 
(1 row)

SELECT pg_eucjp('xa4', 'xca', 'xca');
 pg_eucjp 
----------
 (null)
(1 row)

SELECT pg_eucjp('xa0', 'xca');
 pg_eucjp 
----------
 (null)
(1 row)

SELECT pg_eucjp('xa4', 'xff');
 pg_eucjp 
----------
 (null)
(1 row)

SELECT pg_eucjp('x8e', 'xc5');
 pg_eucjp 
----------
 é≈
(1 row)

SELECT pg_eucjp('x8e', 'xc5', 'xc5');
 pg_eucjp 
----------
 (null)
(1 row)

SELECT pg_eucjp('x8e', 'xe0');
 pg_eucjp 
----------
 (null)
(1 row)

SELECT pg_eucjp('x8f', 'xa2', 'xe0');
 pg_eucjp 
----------
 è¢‡
(1 row)

SELECT pg_eucjp('x8f', 'xa0', 'xe0');
 pg_eucjp 
----------
 (null)
(1 row)

SELECT pg_eucjp('x8f', 'xa2', 'xff');
 pg_eucjp 
----------
 (null)
(1 row)

DROP EXTENSION pg_cheat_funcs;
